internal/iapl/policy.go

```diff
@@ -257,20 +399,205 @@ func (v *policy) expandActionBindings() {
 	}
 }
 
+// expandRole creates a list of all permissions, and a resource containing
+// a list of relationship to all permissions.
+func (v *policy) expandRole() {
+	// 1. create a list of all permissions
+	perms := make(map[string]struct{})
+
+	for _, action := range v.ac {
+		perms[action.Name] = struct{}{}
+	}
+
+	for perm := range perms {
+		v.permissions = append(v.permissions, perm)
+	}
+
+	// 2. create a relationship for role owners
+	roleOwners := Relationship{
+		Relation:    RoleOwnerRelation,
+		TargetTypes: make([]types.TargetType, len(v.p.RBAC.RoleOwners)),
+	}
+
+	for i, owner := range v.p.RBAC.RoleOwners {
+		roleOwners.TargetTypes[i] = types.TargetType{Name: owner}
+	}
+
+	// 3. create a list of relationships for all permissions
+	permsRel := make([]Relationship, len(perms))
+
+	for i, perm := range v.permissions {
+		targettypes := make([]types.TargetType, len(v.p.RBAC.RoleRelationshipSubjects))
+
+		for j, subject := range v.p.RBAC.RoleRelationshipSubjects {
+			targettypes[j] = types.TargetType{Name: subject, SubjectIdentifier: "*"}
+		}
+
+		permsRel[i] = Relationship{
+			Relation:    perm + "_rel",
+			TargetTypes: targettypes,
+		}
+	}
+
+	// 4. create a role containing all the relationships shown above
+	var role ResourceType
```

Comment:
This isn't defining a role - it's defining the role type. This should be made clear.